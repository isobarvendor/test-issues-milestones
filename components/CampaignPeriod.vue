<template>
  <div class="container campaign-period-section redbox-withwhiteborder ">
    <v-row no-gutters>

      <v-col
        cols="12"
        sm="12"
        md="4"
      >

          <v-row no-gutters class="logo-title">
               <v-col
                  cols="4"
                  md="12"
                  sm="4"
                >

                  <img src="/img/landing/COKE STUDIO.png" class="logo"  />
                </v-col>

                <v-col
                  cols="8"
                  md="12"
                  sm="8"
                  class="participate-text-container"
                >
                 <div class="tagline" v-html="howData.title"></div>
                </v-col>
          </v-row>

      </v-col>
      <v-col
        cols="12"
        sm="12"
        md="8"
        class="desc-container"
      >
          <div class="periode-container">
            <p>{{this.data.Title}}</p>
            <p class="campaign-period" v-html="startdate+' â€“ '+enddate"></p>
          </div>
          <div class="period-desc-container">
              {{periodDescription}}
          </div>

           <v-row no-gutters   >
               <v-col
                  cols="12"
                  md="4"
                  sm="12"
                   v-for="(item, idx) in howData.slides" :key="'steps'+idx"
                >
                    <v-row no-gutters  class="center-layout" >
                      <v-col
                          cols="5"
                          md="12"
                          sm="5"
                          class="participate-image-container"
                        >
                          <img v-if="$mq == 'sm'&&item.mobileImage.length>0" :src="item.mobileImage[0].url" class="participate-image" />
                          <img v-else-if="item.desktopImage.length>0&&$mq !== 'sm'" :src="item.desktopImage[0].url" class="participate-image" />
                          <img v-else :src="'/img/landing/drink.png'" />
                        </v-col>

                        <v-col
                          cols="7"
                          md="12"
                          sm="7"
                          class="participate-text-container"
                        >
                         <span v-html="item.description" />

                        </v-col>
                  </v-row>

                </v-col>

          </v-row>
            <div v-if="loginInfo" v-html="howData.description" class="description"></div>

      </v-col>
    </v-row>



  </div>
</template>

<script>
import moment from "moment";
import {translation} from "@/constants/index"
moment.locale('en', {
    ordinal: num => {
        const b = num % 10;
        const output = (~~(num % 100 / 10) === 1) ? 'th' :
            (b === 1) ? 'st' :
                (b === 2) ? 'nd' :
                    (b === 3) ? 'rd' : 'th';
        return num + '<sup>' + output + '</sup>';
    },
});

export default {
   name: "CampaignPeriod",
  data() {
    return {
      periodDescription:translation.periodDescription
    };
  },
  computed:{
    startdate (){
        return this.data && moment(this.data.fromDate).format("Do MMMM YYYY")
    },
    enddate (){
      return this.data && moment(this.data.toDate).format("Do MMMM YYYY");
    },
    loginInfo(){
       return this.$store.state.login;
    }


  }
  ,
  props: {
    data: null,
    howData:{}
  },
  beforeMount() {

  }
};
</script>

<style>
.campaign-period-section .description{
  text-align: center;
}
</style>
