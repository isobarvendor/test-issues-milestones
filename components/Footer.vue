<template>
  <footer>
      <div class="container">
        <p v-html="footerText"></p>
        <p>
          <a href="">Privacy Policy</a>
          <a href="">Terms and conditions</a>
        </p>
      </div>
    </footer>
</template>

<script>
export default {
    name:"Footer",
    props:{
      data:null
    },
    data(){
      return {
        footerText:''
      }
    },
    mounted(){
      var text = this.data.footerText;
      this.footerText = text.replace(/The Coca-Cola Company/g, 'The&nbsp;Coca&#8209;Cola&nbsp;Company');
    }
}
</script>

<style>

</style>