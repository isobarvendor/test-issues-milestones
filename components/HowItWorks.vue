<template>
  <div class="container campaign-section">
    <div class="header">{{data.title}}</div>
    <div class="howitworks">
      <div v-for="(item, idx) in data.slides" :key="'steps'+idx" class="howitworks-item">
        <div class="howitworks-number">{{idx+1}}</div>
        <img v-if="$mq == 'sm'" :src="item.mobileImage.url" />
        <img v-else :src="item.desktopImage.url" />
        <div>{{item.description}}</div>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  props: {
    data: Object,
  },
};
</script>

<style>
</style>