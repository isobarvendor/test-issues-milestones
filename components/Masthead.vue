<template>
  <div id="masthead" class="container" >
    <mq-layout :mq="['sm']">
      <svg
        ref="coke_masthead_sm"
        id="coke_masthead_sm"
        data-name="coke_masthead"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
        viewBox="0 0 360 429"
      >
        <def>
          <mask id="clipMask">
            <path
              id="coke_line"
              data-name="coke_line"
              fill="white"
              d="M0,440.7l0-59V0H360v321.8h0l0,84.9c-10.1-1.7-20.3-2.6-30.5-2.6c-36.5,0-72,12.7-111,32.1
			c-23.9,11.9-55.6,22.8-98,22.8C82.3,459,38.7,452,0,440.7z"
            />
          </mask>
        </def>
        <image
          ref="coke_masthead_sm_image"
          :href="this.imgMob"
          preserveAspectRatio="xMidYMid slice"
          mask="url(#clipMask)"
        />
      </svg>
    </mq-layout>
    <mq-layout :mq="['md', 'lg']">
      <svg
        ref="coke_masthead_md"
        id="coke_masthead_md"
        data-name="coke_masthead"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
        viewBox="0 0 834 685"
      >
        <def>
          <mask id="clipMask">
            <path
              id="coke_line"
              data-name="coke_line"
              fill="white"
              d="M0,852.73,0,716.29V0H833.88V577.19H834l-.08,196.63a412.51,412.51,0,0,0-70.61-6c-84.65,0-166.8,29.44-257.07,74.38C451,869.75,377.33,895,279.15,895,190.73,895,89.63,878.72,0,852.73Z"
            />
          </mask>
        </def>
        <image
          ref="coke_masthead_md_image"
          :href="this.imgTab"
          preserveAspectRatio="xMidYMid slice"
          mask="url(#clipMask)"

        />
      </svg>
    </mq-layout>
    <mq-layout mq="xl">
      <svg
        id="coke_masthead"
        data-name="coke_masthead"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
        viewBox="0 0 1280 650"
      >
        <def>
          <mask id="clipMask">
            <path
              id="coke_line"
              data-name="coke_line"
              fill="red"
              d="M0,585.14.05,449.31H0V0H1279.88V192.59h.12l-.12,271.46a634.78,634.78,0,0,0-108.37-9.17c-129.91,0-256,45.19-394.54,114.14-84.84,42.23-197.86,81-348.55,81C292.72,650,137.56,625,0,585.14Z"
            />
          </mask>
        </def>
        <image
          ref="coke_masthead_image"
          :href="this.imgDesk"
          preserveAspectRatio="xMidYMid slice"
          mask="url(#clipMask)"
        />
      </svg>
    </mq-layout>
    <div class="masthead-content">
      <h1>{{data.title}}</h1>
      <p>{{data.description}}</p>
      
    </div>
    <MastheadCountDown v-if="isCountDown" :data="data.endDate" />
  </div>
</template>

<script>
import MastheadCountDown from '../components/MastheadCountDown'
export default {
  name: "Masthead",
  props: {
    data: null,
    isCountDown:null
  },
  components:{
    MastheadCountDown
  },
  data() {
    return {
      imgDesk: this.data.homepage.mastheadSection.desktopImage.length>0 ? this.data.homepage.mastheadSection.desktopImage[0].url : "/img/landing/Hero Banner.png", 
      imgTab: this.data.homepage.mastheadSection.tabletImage.length>0 ? this.data.homepage.mastheadSection.tabletImage[0].url: "/img/landing/Hero Banner.png", 
      imgMob: this.data.homepage.mastheadSection.mobileImage.length>0 ? this.data.homepage.mastheadSection.mobileImage[0].url :"/img/landing/banner_mobile.png",
    }
  },
  mounted() {
    // this.loadMasthead()
    // window.addEventListener('resize', this.loadMasthead)
  },
  methods: {
    /*loadMasthead() {
      this.$refs["coke_masthead_image"].setAttribute(
          "xlink:href",
          this.data.desktopImage[0].url
        );

      if (this.$refs["coke_masthead_md_image"])
        this.$refs["coke_masthead_md_image"].setAttribute(
          "xlink:href",
          this.data.tabletImage[0].url
        );
      if (this.$refs["coke_masthead_sm_image"])
        this.$refs["coke_masthead_sm_image"].setAttribute(
          "xlink:href",
          this.data.mobileImage[0].url
        );
    }*/
  },
  mounted() {

  },
};
</script>

<style>
 #masthead{
      /* background:#de0a1c; */
      padding: 20px;
    }
  @media only screen and (max-width: 600px) {
     #masthead{
      padding: 10px;
    }
  }
    @media only screen and (max-width: 1099px) {
     #masthead{
      padding: 10px;
    }
  }
</style>
