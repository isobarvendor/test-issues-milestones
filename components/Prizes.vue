<template>
  <div class="container campaign-section" v-if="data">
    <div class="header">{{data.title}}</div>
    <div v-html="data.description"></div>
    <div class="container">
      <swiper class="swiper prize-swiper" :options="swiperOption">
        <swiper-slide
          v-for="(item, index) in data.exclusizePrizesSection"
          :key="'prize'+index"
          class="prize-item"
        >
          <img
            v-if="$mq == 'sm'"
            :src="item.image ? item.image[0].url : '/img/prize-default.svg'"
          />
          <img v-else :src="item.image? item.image[0].url : '/img/prize-default.svg'" />
          <div class="prize-description">{{item.title}}</div>
        </swiper-slide>
        <div class="prize-swiper-pagination" slot="pagination"></div>
      </swiper>
    </div>
    <div v-if="winners.luckyWinnerSection" class="winners-section">
      <div class="header">{{winners.title}}</div>
      <a class="button" href="/winners">{{winners.ButtonTitle}}</a>
    </div>
  </div>
</template>

<script>
import { Swiper, SwiperSlide } from "vue-awesome-swiper";
export default {
  components: {
    Swiper,
    SwiperSlide,
  },
  data() {
    return {
      swiperOption: {
        slidesPerView: 2,
        spaceBetween: 10,
        breakpoints: {
          // when window width is >= 320px
          768: {
            slidesPerView: 3,
            spaceBetween: 20,
          },
          992: {
            slidesPerView: 3,
            spaceBetween: 30,
          },
        },
        freeMode: true,
        pagination: {
          el: ".prize-swiper-pagination",
          clickable: true,
          bulletClass: "prize-swiper-pagination-bullet",
          bulletActiveClass: "prize-swiper-pagination-bullet-active",
        },
      },
    };
  },
  props: {
    data: null,
    winners: null,
  },
  beforeMount() {},
};
</script>
<style scoped lang="scss">
.campaign-section{
.winners-section{
  .button{
    color: #333333;
    background: #ffffff;
    font-size: 14px;
    width: 370px;
    margin:auto;
    display: table;
    margin-top: 35px;
    padding: 20px 90px;
  }
}
@media only screen and (max-width: 1199px) {
  .winners-section{
    .button{
      width:315px;
    }
  }
}
}
</style>
