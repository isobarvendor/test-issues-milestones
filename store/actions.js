
import GeneralAPI from '../api/general';

import {LOGIN, SIGNUP, GET_ACCOUNT} from './action_types';

export default {
  [LOGIN]: ({ commit, state, getters }, data) => {
    return new Promise((resolve, reject) => {
      const moduleState = state;
      if (moduleState.login) {
        return resolve(moduleState.login)
      } else {
        GeneralAPI.login(data)
        .then(response => {
          commit('SET_TOKEN', response.data.token);
          sessionStorage.token=response.data.token;
          return resolve(messages);
        })
        .catch(error => {

          return reject(error);
        });
      }
    });
  },
  [GET_ACCOUNT]: ({ commit, state, getters }, token) => {
    return new Promise((resolve, reject) => {
      const moduleState = state;
      if (moduleState.login) {
        return resolve(moduleState.login)
      } else {
        GeneralAPI.getAccount(token)
        .then(response => {

          commit('SET_LOGIN_ACCOUNT', response.data);
          return resolve(messages);
        })
        .catch(error => {

          return reject(error);
        });
      }
    });
  },
  [SIGNUP]: ({ commit, state, getters }, data) => {
    return new Promise((resolve, reject) => {
      const moduleState = state;
      if (moduleState.login) {
        return resolve(moduleState.login)
      } else {
        GeneralAPI.signup(data)
        .then(messages => {
          //commit('SET_LOGIN_ACCOUNT', messages);
          return resolve(messages);
        })
        .catch(error => {
          console.error(error);
          return reject(error);
        });
      }
    });
  },


  /*[SET_GLOBAL_ERROR_DIALOG]:({commit, state}, status)=>{
    commit('SET_GLOBAL_ERROR_DIALOG', status);
  }
  */
};
