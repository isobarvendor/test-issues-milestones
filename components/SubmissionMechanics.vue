<template>
 <div style="width:100%">
  <div class="container campaign-section redbox-withwhiteborder submission-section" id="submission-section" v-if="cmsData&&!submitted">
      <div class="divider">
        <img src="/img/landing/coke line divider.png" width="100%" />
      </div>
      <Login :social="dataSocial" v-if="!this.$store.state.login"   />
      <Form :data="dataForm" :cmsData="cmsData[0]"  v-if="this.$store.state.login" @submit="submit"  />
      <!--Login :social="dataSocial" v-if="false"   />
      <Form :data="dataForm" :cmsData="cmsData[0]" /-->
  </div>
  <div  v-else >
  <div class="container  prize-chance black-red-border" id="prize-chance">

        <div class="wrapper"  v-for="(item, idx) in prize" :key="idx"  >
          <PrizeItem :prize="item" :themes="themes" @playAgain="playAgain" @submitPrize="submitPrize" v-if="prize.length>0" :loading="loading"  />
          <div class="error-message" v-if="idx==prize.length-1">{{errorMessage}}</div>
        </div>
        <div class="error-message" v-if="prize.length==0">{{errorMessage}}</div>
      </div>
      <div v-if="prize.length==0" style="text-align:center">
            <span v-html="thankYouMessage"></span>
             <div class="prize-button-area center" style="margin-top:40px;">
                  <v-btn @click="playAgain">{{submissionText.participateAgain}}</v-btn>
            </div>
      </div>
    <div class="container prize-chance redbox-withwhiteborder joox-section" v-if="prize.length>0 && prize[0].havejoox"  >
      <div class="background-image-joox">
      <img src="/img/landing/back-dots.png" />
    </div>
    <div class="desc-joox" >
        <span v-html="jooxMessage"></span>
        <!--div class="joox-listen">
            <a :href="listenNowLink" target="_blank"><v-btn >{{submissionText.listenNow}}</v-btn></a>
        </div-->
    </div>
  </div>

 </div>
 </div>
</template>

<script>
import {SUBMIT_FORM } from '@/store/action_types';
import Login from './SubmissionLogin'
import Form from './SubmissionForm'
import {translation} from "@/constants/index"

let configID=process.env.configID.split(",");
export default {
  name: "SubmissionMechanics",
  components:{
    Login,
    Form
  },
   props: {
        dataForm: null,

      },
  data() {
    return {
        prize:[],
        submitted:false,
        submitNumber:0,
        listenNowLink:'',
        request:{},
        attemptData:null,
        lotID:null,
        themes:1,
        thankYouMessage:"",
        request:{
          email:''
        },
        errorMessage:null,
        submissionText:translation.submissionText,
        loading:false
    }

  },
    computed:{
    campaignType(){
      return this.dataForm;
    },
    cmsData(){
      return this.$store.getters.getCMSContent;
    },

    email(){
      return this.request.email;
    },
    dataSocial() {
      return {
        email: this.dataForm .socialMedia.Email,
        facebook: this.dataForm.socialMedia.Facebook,
        google: this.dataForm.socialMedia.Google,
        line: this.dataForm.socialMedia.Line,
      }
    }
  },

  methods: {
    playAgain(){
      this.submitted=false;
      this.prize=[];
    },
    submitPrize(){
      let request= this.request;
      let response = {};
      let data = {
        attemptData:this.attemptData,
        request:this.request
      }
      request.configurationId=configID[this.submitNumber-1];

    if(this.submitNumber==1){
      this.loading=true;
      this.$store.dispatch(SUBMIT_FORM,request)
      .then((response)=>{
           data.response=response.data
           this.submitOne(data,true,2);
             this.loading=false;

      }).catch((error)=>{
       if(error.response){
          this.errorMessage=this.submissionText.errorAPI;
        }
        if(error.response&&error.response.data.trace && error.response.data.trace.errorCode=='1'){
              this.errorMessage=this.submissionText.errorPinCode1;
            }
        if(error.response&&error.response.data.trace && error.response.data.trace.errorCode=='2'){
          this.errorMessage=error.response.data.trace.message;
        }

        if(error.response&&error.response.data.trace && error.response.data.trace.errorCode=='4'){
          this.errorMessage=this.submissionText.errorPinCode2;
        }
          if(error.response&&error.response.data.trace && error.response.data.trace.errorCode=='6'){
          this.errorMessage=this.submissionText.errorPinCode3;
        }
          this.loading=false;
      })
    }else{
        this.submitPrizeDouble(3);
    }

    },

     submitPrizeDouble(page=1){
       this.loading=true;
      let request= this.request;
      request.configurationId=configID[0];
      let data = {
        attemptData:this.attemptData,
        request:request
      }

     /* let response={};
      let response2={};
       let result={
            "burnResult": [
                {
                    "pincode": "0J6TC2VP5",
                    "programId": "453269",
                    "lotId": "445580",
                    "burned": true
                }
            ],
            "participationId": "1ylazdy9kkxmpa53",
            "instantWinResult": {
                "winner": true,
              "redeemedPrize": {
                    "prizeId": "kkqhy9so",
                    "voucherCode": "Voucher891",
                    "status": "claimed",
                    "expiryDate": 1620018590000,
                    "name": "JOOX VIP PASS",
                    "shortDescription": "http://google.com",
                    "redeemDescription": "http://abcd",
                    "imgUrl": "/img/landing/week 1 prize.png",
                    "barcodeType": 1,
                    "emailSent": false,
                    "emailMessage": "Email could not be sent"
                }
            }
        }
        response.data=result;*/
       this.$store.dispatch(SUBMIT_FORM,request)
      .then((response)=>{
           data.response=response.data
           this.submitOne(data,false,page);
           request.configurationId=configID[1];

             this.$store.dispatch(SUBMIT_FORM,request)
            .then((response2)=>{
                data.response=response2.data
               this.submitTwo(data,page);
                 this.loading=false;
             }).catch((error)=>{
               if(error.response){
                  this.errorMessage=this.submissionText.errorAPI;
                }
                if(error.response&&error.response.data.trace && error.response.data.trace.errorCode=='1'){
                this.errorMessage=this.submissionText.errorPinCode1;
              }
              if(error.response&&error.response.data.trace && error.response.data.trace.errorCode=='2'){
          this.errorMessage=error.response.data.trace.message;
        }
              if(error.response&&error.response.data.trace && error.response.data.trace.errorCode=='4'){
                this.errorMessage=this.submissionText.errorPinCode2;
              }
                if(error.response&&error.response.data.trace && error.response.data.trace.errorCode=='6'){
                this.errorMessage=this.submissionText.errorPinCode3;
              }
                this.loading=false;
            })
     }).catch((error)=>{
           if(error.response){
          this.errorMessage=this.submissionText.errorAPI;
        }
         if(error.response&&error.response.data.trace && error.response.data.trace.errorCode=='1'){
                this.errorMessage=this.submissionText.errorPinCode1;
              }
              if(error.response&&error.response.data.trace && error.response.data.trace.errorCode=='2'){
          this.errorMessage=error.response.data.trace.message;
        }
              if(error.response&&error.response.data.trace && error.response.data.trace.errorCode=='4'){
                this.errorMessage=this.submissionText.errorPinCode2;
              }
                if(error.response&&error.response.data.trace && error.response.data.trace.errorCode=='6'){
                this.errorMessage=this.submissionText.errorPinCode3;
              }
            this.loading=false;

      })

    },


     submit(data){
      // console.log(data);
      this.submitted=true;
      let response=data.response;
      let lotID=response.lot.lotId;
      this.attemptData =data.attemptData;
      this.submitNumber=1;
      this.lotID=lotID;
      this.request=data.request;
      if(!this.$config.lotID.includes(lotID)){
        this.submitPrizeDouble();
      }else{

          let prize = [
            {
                text : "<h2>"+this.submissionText.textPage+"/h2>"
                ,name:null
                ,note:null
                ,image:null
                ,havejoox:false
                ,button:[{
                    text:this.submissionText.startRedeeming,
                    type:"submission",
                    id:1
                }],
                code:null,
                subName:null,
                isPlayAgain:false
              },
          ];
          this.prize=prize;
       }


    },
    submitOne(data,button=true,page=1){
      this.submitted=true;
      let prizewin=data.response;
      let attemptData =data.attemptData;
      this.submitNumber=2;
      this.request=data.request;
       // console.log(data)
          let prize =[
              {
                  text : attemptData.FormHeading.thankYouMessage,
                  name : prizewin.instantWinResult.redeemedPrize.name,
                  image: prizewin.instantWinResult.redeemedPrize.imgUrl ? prizewin.instantWinResult.redeemedPrize.imgUrl : '/img/landing/week 1 prize.png' ,
                  note : null
                  ,button:button ? [{
                      text:this.submissionText.redeemNextPrize,
                     type:"submission",
                     id:page
                  }]:[]
                  ,havejoox:false,
                  code: null,
                subName:null,
              }
          ];
          this.prize=prize;
         // console.log(prize);
         //  this.listenNowLink=prizewin.instantWinResult.redeemedPrize.redemptionLink;


    },
     submitTwo(data,page=1){
       // console.log(data)
      this.submitted=true;
      let prizewin=data.response;
      let attemptData =data.attemptData;
      this.submitNumber=3;

      let prize =[
              {
                  text :null,
                  name : prizewin.instantWinResult.redeemedPrize.name,
                  image: prizewin.instantWinResult.redeemedPrize.imgUrl ? prizewin.instantWinResult.redeemedPrize.imgUrl : '/img/landing/week 1 prize.png' ,
                  note : null
                  ,button:[{
                      text:this.submissionText.redeemPrize,
                      link:prizewin.instantWinResult.redeemedPrize.redeemDescription + "?"+this.$config.voucherParameter+"="+prizewin.instantWinResult.redeemedPrize.voucherCode,
                      id:page
                  }]
                  ,havejoox:false,
                  code:  prizewin.instantWinResult.redeemedPrize.voucherCode,
                subName:null,
                isPlayAgain:false
              }
          ];

     //console.log(prize)
      this.prize=[...this.prize,...prize]

     // this.listenNowLink=prizewin.instantWinResult.redeemedPrize.shortDescription;

    }
  },
}
</script>

<style>

  .error-message{
    padding-top: 20px;
    color:#fff;
    text-align: center;
  }
 .prize-chance.joox-section{
     padding-top: 80px;
     text-align: center;
    min-height: 380px !important;
 }
.prize-chance .background-image-joox{
    position: absolute;
    top:0px;

    z-index: 0;
}
.prize-chance .background-image-joox img{
    width:100%;
}
.prize-chance .wrapper{
  padding-bottom: 50px;
  padding-top: 30px;
}
.desc-joox{
    z-index: 1;
}
.desc-joox h4{
  font-size:25px;
  max-width: 500px;
  margin: auto;
  padding-bottom: 10px;
  text-align: center;
}
.joox-listen{
    padding-top: 50px;
}
.joox-listen .v-btn:not(.v-btn--round).v-size--default{
    padding:30px;
    font-weight: bold;
}


</style>
