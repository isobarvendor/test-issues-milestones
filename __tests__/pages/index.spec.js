import { mount,  createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';

import Index from '@/pages/index.vue';
import CampaignPeriod from '@/components/CampaignPeriod'
import Prizes from '@/components/Prizes'
import HowItWorks from '@/components/HowItWorks'
import SubmissionMechanics from '@/components/SubmissionMechanics'
import Footer from '@/components/Footer'
import Masthead from '@/components/Masthead'

import {mockCMSConfig,mockCMSContent} from "@/__mock__/mock";

import actions from '@/store/actions';
import mutations from '@/store/mutations';


const $route = {
  name: "index"
}

let localVue, store;


  beforeEach(() => {
    localVue = createLocalVue();
    localVue.use(Vuex);
    store = new Vuex.Store({
      getters: {
        getCMSContent() {
          return mockCMSContent;
        },
        getLoginAccount(){
          return {
            name:"ron",
            email:"ron@dsw.com",
            imageUrl:"tes"
          }
        },
        getCMSConfig(){
          return mockCMSConfig[0];
        }
      },
      state:{
        token:"32432432",
        language: "en",
        listWallet:{
          walletStatus:[
            {amount :30}
          ]
        },

      },
      actions,
      mutations
    });
  });

  const factory = (route) => {
    return mount(Index, {
      mocks: {
        $route : route
      },
      localVue, store,
    });
  };



describe('index', () => {

  test('renders a Masthead', () => {
    const wrapper = factory($route);
    const component= wrapper.findComponent(Masthead);
    expect(component.exists()).toBeTruthy();
  });
  test('renders a CampaignPeriod', () => {
    const wrapper = factory($route);
    const campaignPeriod = wrapper.findComponent(CampaignPeriod);
    expect(campaignPeriod.exists()).toBeTruthy();
  });
  test('renders a Prizes', () => {
    const wrapper = factory($route);
    const prizes = wrapper.findComponent(Prizes);
    expect(prizes.exists()).toBeTruthy();
  });

  test('renders a HowItWorks', () => {
    const wrapper = factory($route);
    const component= wrapper.findComponent(HowItWorks);
    expect(component.exists()).toBeTruthy();
  });

  test('renders a SubmissionMechanics', () => {
    const wrapper = factory($route);
    const component= wrapper.findComponent(SubmissionMechanics);
    expect(component.exists()).toBeTruthy();
  });

  test('renders a Footer', () => {
    const wrapper = factory($route);
    const component= wrapper.findComponent(Footer);
    expect(component.exists()).toBeTruthy();
  });


  test("mounts properly", () => {
    const wrapper = factory($route);
    expect(wrapper.vm).toBeTruthy();
  });

  test("renders properly", () => {
    const wrapper = factory($route);
    expect(wrapper.html()).toMatchSnapshot();
  });

});
