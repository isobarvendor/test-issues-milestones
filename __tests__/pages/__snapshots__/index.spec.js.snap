// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index renders properly 1`] = `
<div id="main" class="home">
  <div class="wrapper">
    <div id="masthead" class="container">
      <div mq="sm"><svg id="coke_masthead_sm" data-name="coke_masthead" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" viewBox="0 0 360 459">
          <def>
            <mask id="clipMask">
              <path id="coke_line" data-name="coke_line" fill="white" d="M0,440.7l0-59V0H360v321.8h0l0,84.9c-10.1-1.7-20.3-2.6-30.5-2.6c-36.5,0-72,12.7-111,32.1
			c-23.9,11.9-55.6,22.8-98,22.8C82.3,459,38.7,452,0,440.7z"></path>
            </mask>
          </def>
          <image href="https://test-015665-java-cloudfronts3content-d25392n5j4ow.s3.ap-southeast-1.amazonaws.com/dummy_masthead_sm_0808a876c7_9f7402fae7.jpeg" preserveAspectRatio="xMidYMid slice" xlink:href="http://www.w3.org/1999/xlink" mask="url(#clipMask)"></image>
        </svg></div>
      <div mq="md,lg"><svg id="coke_masthead_md" data-name="coke_masthead" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" viewBox="0 0 834 895">
          <def>
            <mask id="clipMask">
              <path id="coke_line" data-name="coke_line" fill="white" d="M0,852.73,0,716.29V0H833.88V577.19H834l-.08,196.63a412.51,412.51,0,0,0-70.61-6c-84.65,0-166.8,29.44-257.07,74.38C451,869.75,377.33,895,279.15,895,190.73,895,89.63,878.72,0,852.73Z"></path>
            </mask>
          </def>
          <image href="https://test-015665-java-cloudfronts3content-d25392n5j4ow.s3.ap-southeast-1.amazonaws.com/dummy_masthead_md_55bf2e6985_c5c36c18de.jpeg" preserveAspectRatio="xMidYMid slice" xlink:href="http://www.w3.org/1999/xlink" mask="url(#clipMask)"></image>
        </svg></div>
      <div mq="xl"><svg id="coke_masthead" data-name="coke_masthead" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" viewBox="0 0 1280 650">
          <def>
            <mask id="clipMask">
              <path id="coke_line" data-name="coke_line" fill="red" d="M0,585.14.05,449.31H0V0H1279.88V192.59h.12l-.12,271.46a634.78,634.78,0,0,0-108.37-9.17c-129.91,0-256,45.19-394.54,114.14-84.84,42.23-197.86,81-348.55,81C292.72,650,137.56,625,0,585.14Z"></path>
            </mask>
          </def>
          <image href="https://test-015665-java-cloudfronts3content-d25392n5j4ow.s3.ap-southeast-1.amazonaws.com/sprite_d_masthead_30c1f074ac.jpeg" preserveAspectRatio="xMidYMid slice" xlink:href="http://www.w3.org/1999/xlink" mask="url(#clipMask)"></image>
        </svg></div>
      <div class="masthead-content">
        <h1></h1>
        <p></p>
      </div>
    </div>
    <div class="container campaign-section">
      <p>Participation period</p>
      <p class="campaign-period">Nov 1, 2020 – Nov 30, 2020</p>
    </div>
    <div class="container campaign-section">
      <div class="header">Exclusive Prizes</div>
      <div><a href="#">Lucky Draw Terms &amp; Condition</a></div>
      <div class="container">
        <div class="swiper-container swiper prize-swiper swiper-container-initialized swiper-container-horizontal swiper-container-free-mode">
          <div class="swiper-wrapper" style="webkit-transition-duration: 0ms; transition-duration: 0ms; webkit-transform: translate3d(NaNpx, 0px, 0px); transform: translate3d(NaNpx, 0px, 0px);">
            <div class="prize-item swiper-slide"><img src="https://test-015665-java-cloudfronts3content-d25392n5j4ow.s3.ap-southeast-1.amazonaws.com/image_1_032c59149f.jpeg">
              <div class="prize-description">6 month voucher for Spotify Premium</div>
            </div>
            <div class="prize-item swiper-slide"><img src="https://test-015665-java-cloudfronts3content-d25392n5j4ow.s3.ap-southeast-1.amazonaws.com/prize_default_98bd155817.svg">
              <div class="prize-description">Premier League Experience Live In England</div>
            </div>
            <div class="prize-item swiper-slide"><img src="https://test-015665-java-cloudfronts3content-d25392n5j4ow.s3.ap-southeast-1.amazonaws.com/c1_ebbb085e60.png">
              <div class="prize-description">65-inch 4K HDR OLED TV</div>
            </div>
            <div class="prize-item swiper-slide"><img src="https://test-015665-java-cloudfronts3content-d25392n5j4ow.s3.ap-southeast-1.amazonaws.com/c1_ebbb085e60.png">
              <div class="prize-description">65-inch 4K HDR OLED TV</div>
            </div>
          </div>
          <div class="prize-swiper-pagination"></div><span class="swiper-notification" aria-live="assertive" aria-atomic="true"></span>
        </div>
      </div>
      <div class="winners-section">
        <div class="header">COULD YOU BE A LUCKY WINNER?</div> <a href="/winners" class="button">Check winner</a>
      </div>
    </div>
    <div class="container campaign-section">
      <div class="header">How it Works?</div>
      <div class="howitworks">
        <div class="howitworks-item">
          <div class="howitworks-number">1</div> <img src="https://test-015665-java-cloudfronts3content-d25392n5j4ow.s3.ap-southeast-1.amazonaws.com/c1_ebbb085e60.png">
          <div>Buy $20 worth of coca cola</div>
        </div>
        <div class="howitworks-item">
          <div class="howitworks-number">2</div> <img src="https://test-015665-java-cloudfronts3content-d25392n5j4ow.s3.ap-southeast-1.amazonaws.com/c2_4896b9b39f.png">
          <div>Take a photo of your receipt of the purchased Coca Cola</div>
        </div>
        <div class="howitworks-item">
          <div class="howitworks-number">3</div> <img src="https://test-015665-java-cloudfronts3content-d25392n5j4ow.s3.ap-southeast-1.amazonaws.com/c3_8eeb8b1d5d.png">
          <div>Enter your details &amp; upload the receipt in the form below</div>
        </div>
      </div>
    </div>
    <div class="container campaign-section">
      <div class="login-box column center">
        <p class="text-center">Sign in or register a Coca-Cola <br>account to enter the draw</p> <a class="social-button fb"><i class="ic ic-fb"></i><span>Continue with Facebook</span></a>
        <!----> <a class="social-button google"><i class="ic ic-google"></i><span>Continue with Google</span></a>
        <!---->
        <!---->
      </div>
      <!---->
    </div>
  </div>
  <footer>
    <div class="container">
      <p>© 2020 The Coca Cola Company. "COCA COLA" and contour bottle design are trademarks of The Coca Cola Company.</p>
      <p><a href="/privacy">Privacy Policy</a> <a href="/tnc">Terms and conditions</a></p>
    </div>
  </footer>
</div>
`;
